dissect/target/plugins/os/windows/sru.py

```diff
@@ -364,10 +366,13 @@ def check_compatible(self) -> None:
         if not self._sru:
             raise UnsupportedPluginError("No SRUDB found")
 
-    def read_records(self, table_name, record_type):
+    def read_records(
+        self, table_name: str, record_type: TargetRecordDescriptor
+    ) -> Iterator[TargetRecordDescriptor | None]:
```

Comment:
```suggestion
    ) -> Iterator[TargetRecordDescriptor]:
```
It won't iterate values of `None`, it'd just be an empty iterator.

Also `TargetRecordDescriptor` is not quite correct. It's actually an iterator of all the listed record types above... :smile:. Maybe make a global type alias for a union of all of those called `SRURecord`?

Reply:
Added in [a7e8895](https://github.com/fox-it/dissect.target/pull/472/commits/a7e8895d7b07e618cb6e690c44758f66346039ec).